File: src/table.py
class Table:
    _index: list[str]
    _columns: list[str]
    _values: list[list[int or float]]

    _cur_row: int
    _cur_col: int

    def __init__(self, index: list[str], columns: list[str]):
        self._index = index
        self._columns = columns
        self._values = [[0 for _ in range(len(columns))] for __ in range(len(index))]
        self._cur_col = 0
        self._cur_row = 0

    def set(self, value: int) -> None:
        self._values[self._cur_row][self._cur_col] = value
        self._cur_col += 1
        if self._cur_col == len(self._columns):
            self._cur_row += 1
            self._cur_col = 0

    def to_tsv(self, file_name: str) -> None:
        with open(file_name, 'w+', encoding='utf-8') as file:
            file.write('\t' + '\t'.join(self._columns) + '\n')
            for row_name, row in zip(self._index, self._values):
                file.write('\t'.join([row_name] + list(map(str, row))) + '\n')

File: src/Table.java
import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.PrintStream;
import java.util.List;

public class Table {
    String title;
    List<String> index, columns;
    long[][] values;

    int cur_row = 0, cur_col = 0;

    Table(String title, List<String> index, List<String> columns) {
        this.title = title;
        this.index = index;
        this.columns = columns;
        values = new long[index.size()][columns.size()];
    }

    void set(long value) {
        values[cur_row][cur_col] = value;
        ++cur_col;
        if (cur_col == columns.size()) {
            ++cur_row;
            cur_col = 0;
        }
    }

    public void toTsv(String fileName) {
        try {
            PrintStream outputStream = new PrintStream(new FileOutputStream(fileName));
            outputStream.println(title);
            outputStream.println("\t" + String.join("\t", columns));
            outputStream.println(index.get(0) + "\t" + values[0][0]);
            outputStream.flush();
            outputStream.close();
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        }
    }
}
